multi_import/multi_importer.py

```diff
@@ -85,9 +85,20 @@ def import_data(self, data):
 
         bound_importers = self._transform_multi_input(data)
 
-        for importer, filename, dataset in bound_importers:
-            result = importer.import_data(dataset, context, transaction=False)
-            results.add_result(filename, result)
+        for importer, datasets in bound_importers:
+            serializer_context = importer.get_serializer_context(context)
+
+            read_datasets = [
+                (filename, importer.read_rows(dataset))
+                for filename, dataset in datasets
+            ]
+
+            for filename, rows in read_datasets:
+                importer.load_instances(rows, serializer_context)
+
+            for filename, rows in read_datasets:
+                result = importer.process_rows(rows, serializer_context)
+                results.add_result(filename, result)
```

Comment:
OK this sort of makes sense.  Instead of doing the entire reading, loading and processing for each file one at a time, the reading and loading are done for all the files (one at a time) and then only after that is the processing done for one file at a time.  The loading step (specifically `load_instance` on the Importer) will raise an error if more than one of the same thing has been loaded.  

But i guess i'm confused as to why it wasn't previously working.  If we process one file at a time, the second time that 'load_instance' is called to import an item with the same PK, wouldn't that have raised the error anyway?